### PR TITLE
allow enabling eslint

### PR DESCRIPTION
**Pull request by alxnddr**
EsLint loader slows down dev build significantly. It is great to catch mistakes as early as possible, but because they will be caught in the end anyway on our CI, it is worth trying to disable EsLint for development.

If you still need eslint in dev mode just use `export USE_ESLINT=true`. 

**Results I got on my Apple Silicon M1 Max**
Before: 2m11s
After: 18s

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -27,6 +27,9 @@ const BUILD_PATH = __dirname + "/resources/frontend_client";
 const WEBPACK_BUNDLE = process.env.WEBPACK_BUNDLE || "development";
 const devMode = WEBPACK_BUNDLE !== "production";
 const useFilesystemCache = process.env.FS_CACHE === "true";
+const shouldUseEslint =
+  process.env.WEBPACK_BUNDLE !== "production" &&
+  process.env.DISABLE_ESLINT !== "true";
```
Comment (ariya) [inline webpack.config.js:32]: Is there a way (without causing a hardship) to stick a with "positive" name?

Double negative [can be confusing at times](https://ariya.io/2016/04/api-names-and-begging-the-negatives).

Comment (alxnddr) [reply webpack.config.js]: Good point, I renamed it to `USE_ESLINT`

**Review by nemanjaglumac — APPROVED** [review]
🚀

**Review by kulyk — APPROVED** [review]
Fantastic!

**Comment by flamber** [comment]
Can we make sure that it's added to the [docs](https://github.com/metabase/metabase/blob/master/docs/developers-guide/devenv.md#frontend-development), and perhaps we even need to update the [PR message](https://github.com/metabase/metabase/blob/master/.github/PULL_REQUEST_TEMPLATE.md)?

**Review by gusaiani — COMMENTED** [review]
Wdyt of adding something on a Readme file about this possibility?

Case in point: testing locally, the compile is real fast now.
When I open my editor, it no longer lints ES (no surprise here) but editing became really slow.

**Comment by alxnddr** [comment]
> Can we make sure that it's added to the [docs](https://github.com/metabase/metabase/blob/master/docs/developers-guide/devenv.md#frontend-development), and perhaps we even need to update the [PR message](https://github.com/metabase/metabase/blob/master/.github/PULL_REQUEST_TEMPLATE.md)?

Good catch. I've updated the docs. I'm not sure what to change in the PULL_REQUEST_TEMPLATE regarding this change because it still suggests running linters. Overall, I think the PR template is not so helpful, so I'll change it in another PR

**Comment by alxnddr** [comment]
> Wdyt of adding something on a Readme file about this possibility?
> 
> Case in point: testing locally, the compile is real fast now. When I open my editor, it no longer lints ES (no surprise here) but editing became really slow.

I've updated the docs now. I wonder what exactly became slow and which editor do you use?

**Pull request merged**